clio-client/src/main/scala/org/broadinstitute/clio/client/util/FutureWithErrorMessage.scala

```diff
@@ -0,0 +1,20 @@
+package org.broadinstitute.clio.client.util
+
+import com.typesafe.scalalogging.Logger
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+object FutureWithErrorMessage {
+  implicit class FutureWithErrorMessage[A](future: Future[A]) {
+    def withErrorMsg(message: String)(implicit ec: ExecutionContext,
+                                      logger: Logger): Future[A] = {
+      future andThen {
```

Comment:
`andThen` is only for side-effects, it doesn't actually change the contents of the Future. So if you do:
```scala
Future.successful(123).andThen { case Success(n) => n + 2 }
```
You still get `123`.

Since you're not doing anything extra in the `Success` case here, you can change this to:
`future recover { ex => logger.err(message); throw new Exception(message, ex) }`